addons/stock/models/stock_picking.py

```diff
@@ -898,9 +898,16 @@ def _log_activity_get_documents(self, orig_obj_changes, stream_field, stream, so
         # {'(delivery_picking_1, admin)': stock.move(1, 2)
         #  '(delivery_picking_2, admin)': stock.move(3)}
         visited_documents = {}
+        grouped_moves = {}
         if stream == 'DOWN':
             if sorted_method and groupby_method:
-                grouped_moves = groupby(sorted(origin_objects.mapped(stream_field), key=sorted_method), key=groupby_method)
+                grp_moves = groupby(sorted(origin_objects.mapped(stream_field), key=sorted_method), key=groupby_method)
+                for ((parent, responsible), moves) in grp_moves:
+                    if grouped_moves.get((parent, responsible, True)):
+                        grouped_moves[(parent, responsible, True)] |= moves
+                    else:
+                        grouped_moves[(parent, responsible, True)] = moves
+                grouped_moves = grouped_moves.items()
```

Comment:
why?

Reply:
if a user wants to log the note/activity on the following documents, then it is not feasible for the system to make any changes on those records, so the system has to generate an activity always., that's the reason why we forcefully apply activity true here.

Reply:
no, why are you doing all this gymnastic with grp_moves? you don't use grouped_moves in a different way than before afterwards.

you could replace all of this with the use of the orm method "concat" but again i don't see why you need it now and why we didn't need it anymore